spec/rails_helper.rb

```diff
@@ -713,6 +713,7 @@ def using_session_with_localhost_resolution(name, &block)
     unfreeze_time
     ActionMailer::Base.deliveries.clear
     Discourse.redis.flushdb
+    Scheduler::Defer.do_all_work
```

Comment:
We need this to ensure that the queue is empty before we rollback the database transaction for the example.